### PR TITLE
Update API to work with latest transformers/datasets package

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

We shouldn't rely on such an older version for transformers/datasets. This will require re-training the model (which I've done locally, Once approved I can upload the new model weights to S3).

### CODE REVIEW

File: flash/core/classification.py [hunk 1]
```diff
@@ -11,19 +11,22 @@
...
 from flash.core.model import Task
 
 
 class ClassificationDataPipeline(TaskDataPipeline):
 
-    def before_uncollate(self, batch: torch.Tensor) -> torch.Tensor:
+    def before_uncollate(self, batch: Union[torch.Tensor, tuple, SequenceClassifierOutput]) -> torch.Tensor:
         if isinstance(batch, tuple):
             batch = batch[0]
+        elif isinstance(batch, SequenceClassifierOutput):
```
Comment (carmocca) [inline flash/core/classification.py:28]: Shouldn't this be in `TextClassificationPipeline` instead?

Comment (SeanNaren) [reply flash/core/classification.py:28]: I didn't even realise that was a thing, yes it should be!

**Review by teddykoker — APPROVED** [review]
🔥 🔥

File: flash/core/classification.py [hunk 2]
```diff
@@ -11,17 +11,18 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any
+from typing import Any, Union
 
 import torch
+from transformers.modeling_outputs import SequenceClassifierOutput
```
Comment (carmocca) [inline flash/core/classification.py:17]: Unused import

**Review by carmocca — APPROVED** [review]

**Pull request merged**